File: whisper.py
import time

start = time.time()
from transformers import pipeline
transcriber = pipeline(model="openai/whisper-tiny", torch_dtype="auto", chunk_length_s=10)
print(transcriber('audio2.m4a', batch_size=8, return_timestamps=True))
end = time.time()
print(end - start)

File: whisper.js
import {HfInference} from "@huggingface/inference";
import fs from 'fs';
const HF_TOKEN = 'xx';
const inference = new HfInference(HF_TOKEN);

console.log(JSON.stringify(await transcribe('audio2.mp3')));

async function transcribe(path) {
    const response = await inference.automaticSpeechRecognition({
        model: 'openai/whisper-large-v3',
        data: JSON.stringify({
            inputs: fs.readFileSync(path, {encoding: 'base64'}).toString(),
            parameters: {
                return_timestamps: true
            }
        })
    });
    return response;
}
